plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -928,7 +939,8 @@ private static ObjectReadFunction longTimestampWithTimeZoneReadFunction()
         return ObjectReadFunction.of(
                 LongTimestampWithTimeZone.class,
                 (resultSet, columnIndex) -> {
-                    OffsetDateTime offsetDateTime = resultSet.getObject(columnIndex, OffsetDateTime.class);
```

Comment:
Is this causing some performance penalty?

Reply:
Is it possible to perform this conversion only for historical dates ? I'm not sure if adding additional `if` would cause additional penalties ?

Reply:
I believe so, we can get the MSSQL `DateTimeOffset` object and if it is before 1583 we can do the String conversion. Presumably it'd more performant that always doing the string parsing.

Reply:
Added a check to skip the date parsing if it is before 1583.